python/paddle/distributed/auto_parallel/local_map.py

```diff
@@ -0,0 +1,280 @@
+                        f"but got {spec}!"
+                    )
+
+                    if not isinstance(spec, list):
+                        spec = [spec]
+
+                    if arg.placements != spec:
+                        if redistribute_inputs:
+                            # Redistribute to input placements
+                            arg = arg.redistribute(process_mesh, spec)
+                        else:
+                            raise ValueError(
+                                f"Mismatched placements for arg {arg}: "
+                                f"got {arg.placements} but required {spec}! "
+                                "Set redistribute_inputs=True if redistribution "
+                                "is needed."
+                            )
+
+                local_arg = dist.auto_parallel.api.dtensor_to_local(
+                    arg, process_mesh, spec
+                )
+                flat_local_args.append(local_arg)
+            else:
+                if in_placements is not None and not isinstance(
+                    arg, paddle.Tensor
+                ):
+                    spec = in_placements[idx]
+                    assert spec is None, (
+                        f"Expected None placements for non-tensor input {arg} "
+                        f"but got {spec}!"
+                    )
+                flat_local_args.append(arg)
+
+        local_args = pack_sequence_as(args, flat_local_args)
+        out = func(*local_args, **kwargs)
+
+        if seen_dist_tensor:
+            flat_out = flatten(out)
+            out_placements_tuple = (
+                out_placements
+                if isinstance(out_placements, tuple)
+                else (out_placements,)
+            )
+
+            assert len(flat_out) == len(out_placements_tuple), (
+                "local_map requires one PlacementType for each output value, "
+                f"got {len(out_placements_tuple)} placements but expected "
+                f"{len(flat_out)}!"
+            )
+
+            flat_dist_out = []
+            for out, spec in zip(flat_out, out_placements_tuple):
+                if isinstance(out, paddle.Tensor):
+                    assert not _is_distributed_tensor(
+                        out
+                    ), f"Expected dense tensor output but got {type(out)}: {out}"
+
+                    flat_dist_out.append(
+                        dist.auto_parallel.api.dtensor_from_local(
+                            out, process_mesh, spec
+                        )
+                    )
+                else:
+                    assert spec is None, (
+                        f"Expected None placements for non-tensor output {out} "
+                        f"but got {spec}!"
+                    )
+                    flat_dist_out.append(out)
+
+            return pack_sequence_as(out, flat_dist_out)
+        else:
+            return out
+
+    def _is_distributed_tensor(tensor) -> bool:
```

Comment:
```suggestion
    def is_dist_tensor(tensor) -> bool:
```
这是一个很基础的方法，应该放在更公共的地方，方便其它模块复用。

Reply:
Done